src/runtime_src/core/common/api/xrt_error.cpp

```diff
@@ -205,6 +206,27 @@ class error_impl
 public:
   error_impl(const xrt_core::device* device, xrtErrorClass ecl)
   {
+    //Code for new format; Binary array of error structs in sysfs
+    try {
+      std::vector<char> buf = xrt_core::device_query<xrt_core::query::xocl_errors>(device);
+      if (buf.empty())
+        return;
+      const xcl_errors *errors_buf = reinterpret_cast<xcl_errors *>(buf.data());
```

Comment:
Don't use repeated types.   Should be 
`auto error_buf = reinterpret_cast<const xcl_errors*>(buf.data());`

And coding convention is `type* var` but you avoid this convention by using auto.

Reply:
updated